### ISSUE TITLE
`WS001` false positive: "Section title should be preceded by a blank line" when `[discrete]` immediately precedes the heading

### ISSUE THREAD

**Issue by rdmueller** [docToolchain/asciidoc-linter#48]
The `WS001` rule in `whitespace_rules.py` reports `Section title should be preceded by a blank line` whenever a `[discrete]` attribute appears directly before a section heading, even though that is the correct AsciiDoc syntax — a blank line between `[discrete]` and `== Heading` would break the attribute association.

**Minimal reproducer:**
```asciidoc
[%collapsible]
====
[discrete]
== Core Concepts
content here
====
```

**Expected:** no finding (or at most a warning that `[discrete]` is the recommended way to put headings inside example blocks).

**Actual:** `✗ line N: Section title should be preceded by a blank line`

**Root cause:** The whitelist in `whitespace_rules.py` (around line 165) only accepts lines starting with `[.` or `[[` as valid substitutes for a blank line before a section title. Lines starting with `[d` (e.g. `[discrete]`) are not covered.

**Fix suggestion:** Extend the whitelist condition to also accept any line that starts with `[` (i.e. any AsciiDoc block attribute list), or at minimum add `"[discrete"` to the tuple.
